pandas/tools/pivot.py

```diff
@@ -136,8 +136,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
```

Comment:
I understand, but it is more clear simply to give all indexers (the level positioners) here rather than worry about determining if it has a name or not.